### ISSUE TITLE
pbspro wrapper not setting memory requirements

### ISSUE THREAD

**Issue by marchoeppner** [ssadedin/bpipe#158]
Hi,

using Bpipe 0.9.9b1. The wrapper for pbspro does not interpret memory requests. While I don't fully understand the structure of the wrapper, the variable $memory_request never actually gets set, so I assume that is why it goes with a default of 1GB?

Would be great if that could be looked into and possibly fixed!

/M

**Comment by ssadedin** [comment]
Looks like it's never been implemented. I just took a stab at it and pushed a commit up to the master branch to add support for it.

Marc, it would be great if you could build from trunk and give it a try as I'm not able to test on a real PBS pro system. At the moment it assumes you are specifying the memory in GB in the bpipe.config, eg:

    memory="2gb"

or just

    memory=2

**Comment by marchoeppner** [comment]
Hi Simon,

memory works now, procs doesn't. I think the issue is that PBSpro doesn't allow (?) multiple "-l" arguments. A common way of specifying cores and RAM would be:

#PBS -l nodes=1:ppn=16,mem=120GB

(1 node, 16 cores, 120GB Ram).  

The wrapper includes this line, which I think was meant to achieve this - again, no idea where $SELECT_STATEMENT comes from:

 if [[ ! -z $SELECT_STATEMENT ]]; then
        select_request="#PBS -l $SELECT_STATEMENT"
    fi

**Comment by ssadedin** [comment]
Thanks, glad I got you to check this!

I will make it consolidate the resources into a single statement.

**Comment by ssadedin** [comment]
I pushed another update. It now combines multiple parameters into a single -l in the form you suggest above. Right now it always requests nodes=1.

I looked at the SELECT_STATEMENT support - that's actually coming straight from a variable from the bpipe.config, "select_statement". So if you set that now, it should get passed through. As it is currently implemented that will end up as a separate -l parameter. From my reading of the user guide, that should work OK but I'm not sure.

Thanks for your testing & let me know if this helps!
